lib/SDIR/Translate/translateToSDFG.cpp

```diff
@@ -0,0 +1,1597 @@
+
+  jemit.startNamedObject("attributes");
+  jemit.startNamedObject("data");
+  jemit.printKVPair("type", "InterstateEdge");
+
+  jemit.startNamedObject("attributes");
+  jemit.startNamedObject("assignments");
+
+  if (op.assign().hasValue()) {
+    ArrayAttr assignments = op.assign().getValue();
+
+    for (Attribute assignment : assignments) {
+      if (StringAttr strAttr = assignment.dyn_cast<StringAttr>()) {
+        StringRef content = strAttr.getValue();
+        std::pair<StringRef, StringRef> kv = content.split(':');
+        jemit.printKVPair(kv.first.trim(), kv.second.trim());
+      } else {
+        mlir::emitError(
+            op.getLoc(),
+            "'assign' must be an ArrayAttr consisting of StringAttr");
+        return failure();
+      }
+    }
+  }
+
+  jemit.endObject(); // assignments
+  jemit.startNamedObject("condition");
+  jemit.printKVPair("language", "Python");
+  if (op.condition().hasValue()) {
+    if (op.condition().getValue().empty()) {
+      jemit.printKVPair("string_data", "1");
+    } else {
+      jemit.printKVPair("string_data", op.condition().getValue());
+    }
+  } else {
+    jemit.printKVPair("string_data", "1");
+  }
+  jemit.endObject(); // condition
+  jemit.endObject(); // attributes
+  jemit.endObject(); // data
+  jemit.endObject(); // attributes
+
+  SDFGNode sdfg = dyn_cast<SDFGNode>(op->getParentOp());
+
+  StateNode srcState = sdfg.getStateBySymRef(op.src());
+  unsigned srcIdx = sdfg.getIndexOfState(srcState);
+  jemit.printKVPair("src", srcIdx);
+
+  StateNode destState = sdfg.getStateBySymRef(op.dest());
+  unsigned destIdx = sdfg.getIndexOfState(destState);
+  jemit.printKVPair("dst", destIdx);
+
+  jemit.endObject();
+  return success();
+}
+
+//===----------------------------------------------------------------------===//
+// AllocOp
+//===----------------------------------------------------------------------===//
+
+LogicalResult printScalar(AllocOp &op, JsonEmitter &jemit) {
+  jemit.startNamedObject(op.getName());
+  jemit.printKVPair("type", "Scalar");
+
+  jemit.startNamedObject("attributes");
+
+  Type element = op.getType().getElementType();
+  if (translateTypeToSDFG(element, op.getLoc(), jemit, "dtype").failed())
+    return failure();
+
+  jemit.startNamedList("shape");
+  jemit.startEntry();
+  jemit.printInt(1);
+  jemit.endList(); // shape
+
+  jemit.printKVPair("transient", "false", /*stringify=*/false);
+  jemit.printKVPair("storage", "Default");
+  jemit.printKVPair("lifetime", "Scope");
+
+  jemit.endObject(); // attributes
+
+  jemit.endObject();
+  return success();
+}
+
+LogicalResult translateAllocToSDFG(AllocOp &op, JsonEmitter &jemit) {
+  if (op.getType().getShape().size() == 0)
+    return printScalar(op, jemit);
+
+  jemit.startNamedObject(op.getName());
+  jemit.printKVPair("type", "Array");
+
+  jemit.startNamedObject("attributes");
+  // TODO: Print the values you can derive
+  // jemit.printKVPair("allow_conflicts", "false", /*stringify=*/false);
```

Comment:
what does this mean?

Reply:
This is a reminder to print as much information as possible. Everything that can be derived should be emitted. The comment does not correspond to the line below it.